api/handlers.go

```diff
@@ -503,7 +507,7 @@ func getContainerJSON(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	resp, err := client.Get(scheme + "://" + container.Engine.Addr + "/containers/" + container.ID + "/json")
+	resp, err := client.Get(scheme + "://" + container.Engine.Addr + "/containers/" + container.ID + "/json" + sizeQuery)
```

Comment:
Actually I like the concept. While I prefer using `InspectContainer(container.ID)` to call docker daemon instead of constructing a new one by hand.
Since if we use `InspectContainer`, then we can edit the field of passed query or something else easier.
Just my thought. Open for input.
And there is a PR related this: https://github.com/docker/swarm/pull/2335/files